pkg/internal/cryptoutil/utils.go

```diff
@@ -50,20 +69,78 @@ func IsChachaKeyValid(key []byte) bool {
 // KeyPair represents a private/public key pair
 type KeyPair struct {
 	// Priv is a private key
-	Priv []byte
+	Priv []byte `json:"priv,omitempty"`
 	// Pub is a public key
-	Pub []byte
+	Pub []byte `json:"pub,omitempty"`
 }
 
-// IsKeyPairValid is a utility function that validates a KeyPair
-func IsKeyPairValid(kp KeyPair) bool {
+// EncKeyPair represents a private/public encryption key pair
+type EncKeyPair struct {
+	KeyPair `json:"keypair,omitempty"`
+	// Alg is the encryption algorithm of keys enclosed in this key pair
+	Alg EncryptionAlgorithm `json:"alg,omitempty"`
+}
+
+// SigKeyPair represents a private/public signature (verification) key pair
+type SigKeyPair struct {
+	KeyPair `json:"keypair,omitempty"`
+	// Alg is the signature algorithm of keys enclosed in this key pair
+	Alg SignatureAlgorithm `json:"alg,omitempty"`
+}
+
+// KeyPairCombo represents a pair of key pairs, one for encryption and one for signature
+// usually stored in a wallet, it helps prevent converting signing keys into encryption ones
+type KeyPairCombo struct {
+	*EncKeyPair `json:"enckeypair,omitempty"`
+	*SigKeyPair `json:"sigkeypair,omitempty"`
+}
+
+// isKeyPairValid is a utility function that validates a KeyPair
+func isKeyPairValid(kp KeyPair) bool {
 	if kp.Priv == nil || kp.Pub == nil {
 		return false
 	}
 
 	return true
 }
 
+// IsEncKeyPairValid is a utility function that validates an EncKeyPair
+func IsEncKeyPairValid(kp *EncKeyPair) bool {
+	if !isKeyPairValid(kp.KeyPair) {
+		return false
+	}
+
+	switch kp.Alg {
```

Comment:
Why would the util package know all valid types? If it does know all valid types, why is a string type used?

Reply:
it's `cryptoutil` package which serves as a utility to tell what algorithms are supported by the framework

`Alg` in `EnKeyPair.KeyPair.Alg` is of type `EncryptionAlgorithm`
while in `SigKeyPair.KeyPair.Alg` is of type `SignatureAlgorithm`

so the comment type for both is string, hence `Alg` type is string

does it make sense to pull `Alg` up? this way it will be properly typed in both `EncKeyPair` and `SigKeyPair`.

Reply:
So this also needs to be refactored.

- A particular crypto implementation will support some set of algorithms.
- From the available crypto implementation, a particular KMS implementation will support some set of algorithms.
- The framework will get it's supported list from Crypto and KMS. Framework should not care about algorithm IDs - it's simply a string.
- A particular package that uses KMS might care to only allow a subset of what's available. (or fail if nothing it can use is available.)

Reply:
sounds good, added these points to #596